File: DSA/LeetCode/Arrays/1894. Find the Student that Will Replace the Chalk.js
/**
 * @param {number[]} chalk
 * @param {number} k
 * @return {number}
 */
var chalkReplacer = function (chalk, k) {


    let sum = chalk.reduce((accm, val) => accm + val);

    if (sum < k) {
        k = k % sum
    }

    let i = 0;
    while (k >= chalk[i]) {
        k -= chalk[i]
        i++
        i = i % chalk.length
    }

    return i;
};

File: DSA/LeetCode/Arrays/1894. Find the Student that Will Replace the Chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        
        chalkSum = sum(chalk)
        if chalkSum < k:
            k =  k%chalkSum
        i = 0

        while chalk[i] <= k:
            k -= chalk[i]
            i +=1
            i = i % len(chalk)
        
        return i
        
        